diff -ur scratch-1.4.0.6.src.orig/src/scratch scratch-1.4.0.6.src/src/scratch
--- scratch-1.4.0.6.src.orig/src/scratch	2011-07-01 18:21:32.000000000 -0400
+++ scratch-1.4.0.6.src/src/scratch	2012-09-10 09:39:20.555278005 -0400
@@ -4,11 +4,12 @@
 # Original Author: Bert Freudenberg
 # Adapted by:      Miriam Ruiz
 
-VM_VERSION=`find /usr/lib/squeak/ -name "squeakvm" -type f|cut -f5 -d"/"`
-SQ_DIR=/usr/lib/squeak/$VM_VERSION
-VM="$SQ_DIR/squeakvm"
-VMOPTIONS="-encoding UTF-8 -vm-display-x11 -xshm -plugins /usr/lib/scratch/Plugins/:$SQ_DIR/"
-IMAGE="/usr/lib/scratch/Scratch.image"
+VM_VERSION=$( squeak -version|head -1 |awk '{print $1}' )
+VM_LIBARCH=$( squeak -version|awk '/^default plugin/ {print $4}' |cut -d'/' -f3 )
+SQ_DIR=/usr/$VM_LIBARCH/squeak/$VM_VERSION
+VM="/usr/bin/squeak"
+VMOPTIONS="-encoding UTF-8 -vm-display-x11 -xshm -plugins /usr/$VM_LIBARCH/scratch/Plugins/"
+IMAGE="/usr/share/scratch/Scratch.image"
 IMOPTIONS=""
 DOCUMENT=""
 WRAPPER=""
