--- scratch-1.4.0.7.src/src/scratch.orig	2012-09-28 15:49:36.000000000 -0400
+++ scratch-1.4.0.7.src/src/scratch	2012-09-28 15:48:42.000000000 -0400
@@ -8,7 +8,7 @@
 VM_LIBARCH=$( squeak -version|awk '/^default plugin/ {print $4}' |cut -d'/' -f3 )
 SQ_DIR=/usr/$VM_LIBARCH/squeak/$VM_VERSION
 VM="/usr/bin/squeak"
-VMOPTIONS="-encoding UTF-8 -plugins /usr/$VM_LIBARCH/scratch/Plugins/"
+VMOPTIONS="-encoding UTF-8 -vm-display-x11 -xshm -plugins /usr/$VM_LIBARCH/scratch/Plugins/"
 IMAGE="/usr/share/scratch/Scratch.image"
 IMOPTIONS=""
 DOCUMENT=""
